internal/model/application.go

```diff
@@ -0,0 +1,69 @@
+package model
+
+import (
+	"github.com/G-Research/yunikorn-core/pkg/webservice/dao"
+)
+
+type Application struct {
+	ModelMetadata          `json:",inline"`
+	dao.ApplicationDAOInfo `json:",inline"`
+}
+
+func (app *Application) MergeFrom(appInfo *dao.ApplicationDAOInfo) {
+	app.Partition = appInfo.Partition
+	app.QueueName = appInfo.QueueName
+	app.SubmissionTime = appInfo.SubmissionTime
+	app.FinishedTime = appInfo.FinishedTime
+	app.State = appInfo.State
+	app.User = appInfo.User
+	app.Groups = appInfo.Groups
+	app.RejectedMessage = appInfo.RejectedMessage
+	app.StateLog = appInfo.StateLog
+	app.HasReserved = appInfo.HasReserved
+	app.Reservations = appInfo.Reservations
+	app.MaxRequestPriority = appInfo.MaxRequestPriority
+	app.StartTime = appInfo.StartTime
+	app.PlaceholderData = appInfo.PlaceholderData
+
+	lookup := make(map[string]struct{})
```

Comment:
I'd suggest `any` instead of `struct{}`

Reply:
struct has 0 size while any has 2 fields in a struct. Since I don't really care about the value, and this is just a set, it is always more readable to me when I see `map[string]struct{}` to know what is it used for.